content/en/docs/Writing policies/mutate.md

```diff
@@ -485,3 +485,51 @@ Labels:       backup-needed=no
               type=database
 <snip>
 ```
+
+## foreach
+
+The `foreach` declaration simplifies mutation of sub-elements in resource declarations, for example Containers in a Pod. 
+
+A `foreach` must contain a `list` attribute that defines the list of elements it processes and a `patchStrategicMerge` declaration. For example, iterating over the list of containers in a Pod is performed using this `list` declaration:
+
+```yaml
+list: request.object.spec.containers
+patchStrategicMerge:
+  spec:
+    containers:
+       ...
```

Comment:
In the PR proof manifest, I see an object is defined directly under `spec.containers`:
```
list: request.object.spec.containers
patchStrategicMerge:
  spec:
    containers:
       image: ...
```
While when I build another mutate policy using `foreach`,  I need to specify `-` to indicate the list value:
```yaml
        foreach:
          list: "request.object.spec.containers"
          patchStrategicMerge:
            spec:
              containers:
              - (name): "*"
              ...
````
What is the expected format?

Reply:
Isn't `-` used for JSON Patch? You can use `element` to reference the list element within the foreach. Is that what you were asking about?

Reply:
This is what a Pod manifest looks like, note `-` under tag `containers` because it takes a container list.
```yaml
...
 spec:
   containers:
   - image: nginx:latest
     name: nginx
```

When writing a `patchStrategicMerge` policy, we need to follow the same pattern of the original Kubernetes resource, which means I need to specify `-` in the mutate pattern in order to apply it to Pod:
```yaml
 patchStrategicMerge:
            spec:
              containers:
              - name: nginx
                image: nginx:blue
              ...
```

However in your mutate example, I see no `-` being specified which indicates the `containers` takes an object (map).
```yaml
        patchStrategicMerge:
          spec:
            containers:
              name: "{{ element.name }}"           
              image: "{{ registry.io/{{images.containers.{{element.name}}.path}}:{{images.containers.{{element.name}}.tag}} }}"
```

So my question is, what should be the expected syntax for mutate foreach policy? For this policy I attached in the [proof manifest](https://github.com/kyverno/kyverno/pull/2507), it only works if I define a list under tag `containers`.